### PR TITLE
[lldp] add lldpRemManAddrTable, lldpLocManAddrTable

### PR DESCRIPTION
**Pull request by mykolaf**
What I did:
- Implemented lldpRemManAddrTable & lldpLocManAddrTable

Also requires https://github.com/Azure/sonic-snmpagent/pull/72

**lldpRemManAddrTable**

OID | SNMP counter | Description
-- | -- | --
1.0.8802.1.1.2.1.4.2.1.1 | lldpRemManAddrSubtype | The type of management address identifier encoding used in the associated 'lldpRemManagmentAddr' object
1.0.8802.1.1.2.1.4.2.1.2 | lldpRemManAddr | The string value used to identify the management address component associated with the remote system. The purpose of this address is to contact the management entity
1.0.8802.1.1.2.1.4.2.1.3 | lldpRemManAddrIfSubtype | The enumeration value that identifies the interface numbering method used for defining the interface number, associated with the remote system
1.0.8802.1.1.2.1.4.2.1.4 | lldpRemManAddrIfId | The integer value used to identify the interface number regarding the management address component associated with the remote system
1.0.8802.1.1.2.1.4.2.1.5 | lldpRemManAddrOID | The OID value used to identify the type of hardware component or protocol entity associated with the management address advertised by the remote system agent

**lldpLocManAddrTable**

OID | SNMP counter | Description
-- | -- | --
1.0.8802.1.1.2.1.3.8.1.1 | lldpLocManAddrSubtype | The type of management address identifier encoding used in the associated 'lldpLocManagmentAddr' object
1.0.8802.1.1.2.1.3.8.1.2 | lldpLocManAddr | The string value used to identify the management address component associated with the local system. The purpose of this address is to contact the management entity.
1.0.8802.1.1.2.1.3.8.1.3 | lldpLocManAddrLen | The total length of the management address subtype and the management address fields in LLDPDUs transmitted by the local LLDP agent.
1.0.8802.1.1.2.1.3.8.1.4 | lldpLocManAddrIfSubtype | The enumeration value that identifies the interface numbering method used for defining the interface number, associated with the local system
1.0.8802.1.1.2.1.3.8.1.5 | lldpLocManAddrIfId | The integer value used to identify the interface number regarding the management address component associated with the local system.
1.0.8802.1.1.2.1.3.8.1.6 | lldpLocManAddrOID | The OID value used to identify the type of hardware component or protocol entity associated with the management address advertised by the local system agent.

### CODE REVIEW

File: src/lldp_syncd/dbsyncd.py [hunk 1]
```diff
@@ -40,16 +45,36 @@ def port_handler(self, key, data):
...
+    def port_table_init(self):
         self.port_table = self.config_db.get_table('PORT')
         # supply LLDP_LOC_ENTRY_TABLE and lldpd with correct values on start
         for port_name, attributes in self.port_table.items():
             self.run_command("lldpcli configure lldp portidsubtype local {} description '{}'"
                              .format(port_name, attributes.get("description", " ")))
 
+    def run(self):
+
+        self.port_table_init()
```
Comment (qiluo-msft) [inline src/lldp_syncd/dbsyncd.py]: del the empty line?

File: src/lldp_syncd/dbsyncd.py [hunk 2]
```diff
@@ -17,6 +21,7 @@ def __init__(self):
         self.config_db.connect()
         logger.info("[lldp dbsyncd] Connected to configdb")
         self.port_table = {}
+        self.man_addr = None
```
Comment (qiluo-msft) [inline src/lldp_syncd/dbsyncd.py]: >man_addr [](start = 13, length = 8)

not used?

File: src/lldp_syncd/dbsyncd.py [hunk 3]
```diff
@@ -1,9 +1,13 @@
 import subprocess
+import re
 from swsssdk import ConfigDBConnector
 
 from sonic_syncd import SonicSyncDaemon
 from . import logger
 
+MGMT_INTERFACE_PATTERN = r"MGMT_INTERFACE*"
```
Comment (qiluo-msft) [inline src/lldp_syncd/dbsyncd.py]: >MGMT_INTERFACE_PATTERN [](start = 0, length = 22)

not used?

File: src/lldp_syncd/dbsyncd.py [hunk 4]
```diff
@@ -40,16 +45,36 @@ def port_handler(self, key, data):
         # update local cache
         self.port_table[key] = data
 
-    def run(self):
+    def man_addr_init(self):
```
Comment (qiluo-msft) [inline src/lldp_syncd/dbsyncd.py]: >man_addr_init [](start = 8, length = 13)

mgmt_addr_init

File: src/lldp_syncd/dbsyncd.py [hunk 5]
```diff
@@ -40,16 +45,36 @@ def port_handler(self, key, data):
...
                              .format(port_name, attributes.get("description", " ")))
 
+    def run(self):
+
+        self.port_table_init()
         # subscribe for further changes
         self.config_db.subscribe('PORT', lambda table, key, data:
                                  self.port_handler(key, data))
 
+        self.man_addr_init()
```
Comment (qiluo-msft) [inline src/lldp_syncd/dbsyncd.py]: >man_addr_init [](start = 13, length = 13)

man_addr_init before subscribe?

**Comment by qiluo-msft** [comment]
>     def parse_update(self, lldp_json):

Could you add or update test cases?

---
Refers to: src/lldp_syncd/daemon.py:192 in 7fcf58c. [](commit_id = 7fcf58cb378b0ef44f2299ce344a78e82c12b768, deletion_comment = False)

**Comment by mykolaf** [comment]
Hi @qiluo-msft , I handled previous comments, can you take a look?

**Comment by qiluo-msft** [comment]
> SONIC_ETHERNET_RE_PATTERN = r'^(Ethernet(\d+)|eth0)$'

It's normal practice to include management port in the LLDP. We checked at least on one vendor switch OS.

---
Refers to: src/lldp_syncd/daemon.py:19 in 4a61ab8. [](commit_id = 4a61ab848e4af5ddccad6b8c3eafc4b95d31d5ee, deletion_comment = True)

File: tests/mock_tables/dbconnector.py [hunk 6]
```diff
@@ -15,20 +15,69 @@ def config_set(self, *args):
...
+
         if db == 0:
             with open(INPUT_DIR + '/LLDP_ENTRY_TABLE.json') as f:
                 db = json.load(f)
                 for h, table in db.items():
                     for k, v in table.items():
                         self.hset(h, k, v)
 
+        elif db == 4:
+            with open(INPUT_DIR + '/CONFIG_DB.json') as f:
```
Comment (qiluo-msft) [inline tests/mock_tables/dbconnector.py]: You missed this file?

Comment (mykolaf) [reply tests/mock_tables/dbconnector.py]: I'm afraid so.

**Pull request merged**